### PR TITLE
IDE setup instructions and comparison

### PR DESCRIPTION
**Pull request by karliss**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/developers-docs/first-time.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/developers-docs.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

**Test plan (required)**

* Make sure there are no warnings when running `make quick`
* Check that all links point to the expected page

**Closing issues**

### CODE REVIEW

**Comment by karliss** [comment]
TODO:
* Make the marking of UI element paths consistent
* Debugger Qt type visualization in VSCode

File: docs/source/developers-docs/ide-setup.rst [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+Qt Creator has a builtin visual form editor but not having it in other IDEs is not a major problem. It is also available as standalone tool called Qt Designer and you can configure the file associations so that .ui files are opened using it. Depending on the .ui file and changes you want to make it is sometimes easier to perform them by editing the .ui file as a text file. .ui files are XML files. Most code editors should have some support for XML highlighting and possibly block folding.
+
+All instructions assume that you have already download cutter source and obtained required dependencies as described in :doc:`../building`
+
+Linux
+-----
+
+On a rolling-release distribution or a somewhat recent version of traditional distribution like Ubuntu 18.04 it should be possible to get all the dependencies from official repository. There might
+be some problems with PySide2 and Shiboken2 but it can be easily disabled and isnt' necessary for most work on Cutter. Don't try to install PySide using pip.
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:24]: isnit?

File: docs/source/developers-docs/ide-setup.rst [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+For a quick test the exact versions of libraries used by Cutter release packages can be obtained from `cutter-deps <https://github.com/radareorg/cutter-deps/releases>`_ but they don't contain debug
+versions of libraries so they are not suitable for more serious Cutter development on Windows.
+
+Qt Creator
+----------
+QT Creator is an open source IDE made by same developers as Qt.
+
+Even though Cutter has qmake project cutter.pro it is recommended to use the CMake project in QTCreator.
+QTCreator support for CMake is as good as qmake one but not all Cutter project configuration options are available in qmake project and in future Cutter qmake project might be removed.
+
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:39]: > made by the same developers of Qt

File: docs/source/developers-docs/ide-setup.rst [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Click "Projects" button on the left side of screen and then select "Build". All the project CMake options are listed and can be edited there in a graphical editor.
+
+Editing Qt .ui files
+~~~~~~~~~~~~~~~~~~~~
+Double clicking a .ui file in a file list opens it inside a visual editor. If you want to make changes that are easier to do by editing .ui file as text - right click the file and select "Open With/Plain Text Editor". Switching from visual form editor back to code editor mode will open the .ui file in read only mode with a warning "This file can only be edited in Design mode". To edit use the same steps as described before.
+
+VSCode
+-------
+`VSCode <https://github.com/Microsoft/vscode>`_ is an open source code editor made by Microsoft.
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:75]: Difference in name is important. One was name of open source source code editor and other is Microsoft controlled trademark that can't be freely used. Now I am not so sure about "VSCode" anymore. "Visual Studio Code" was definitely the later. But which one is "VScode" as some places use just "Code" or "Code OSS".

File: docs/source/developers-docs/ide-setup.rst [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+Pros and Cons
+~~~~~~~~~~~~~
+
+- Large amount of plugins
+- Good fallback mechanism for files that are not directly part of project.
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:81]: I don't think "Highly popular, modern, and user friendly." is a very useful for the list of cons and pros and could apply to almost anything. 

Compared to other IDEs in this page I wouldn't call VSCode the heviest.

File: docs/source/developers-docs/ide-setup.rst [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+Recommended plugins
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:83]: Those are somewhat more subjective and would better be separated in different list.

File: docs/source/developers-docs/ide-setup.rst [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+It is recommended to use the latest Visual Studio version 2019 because it has best CMake integration.
+Older VS versions can be used but CMake integration isn't as good. With those it might be better to generate Visual Studio
+project from CMake project using command-line or cmake-gui and opening the generated Visual Studio project instead of opening
+CMake project directly.
+
+Visual Studio supports many different languages and use-cases. Full installation takes a lot of space. To keep the size minimal during installation
+select only component called "Desktop development with C++". Don't worry too much about missing something.
+Additional components can be later added or removed through the VS installer which also serves as an updater and package manager for Visual Studio components.
+
+Pros and Cons
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:170]: Yes, i was thinking of adding that.

File: docs/source/developers-docs/ide-setup.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (ITAYC0HEN) [reply docs/source/developers-docs/ide-setup.rst:24]: edited

File: docs/source/developers-docs/ide-setup.rst [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+- builtin help for Qt API
+- builtin .ui file editor (Qt Designer - visual form editor)
+- builtin helper for displaying Qt types in the debugger
+- Viewing source files that are not directly part of the project (R2 source code) is somewhat inconvenient.
+- The simplest way of installing on non-Linux operating systems require login with Qt account
+
+Project setup
+~~~~~~~~~~~~~
+Instructions made based on Qt Creator 4.12.4 the steps might slightly differ between the versions.
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:51]: There should probably be a verb after instructions?

File: docs/source/developers-docs/ide-setup.rst [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+
+Project setup
+~~~~~~~~~~~~~
+- ``File/Open Folder`` select the folder in which you cloned Cutter
+- If the recommend plugins are already installed in the corner you will see a popup "Would you like to configure project 'cutter'? Source: CMake Tools (Extension)" Click Yes.
+- If you initially dismissed the configuration window or didn't have the plugins installed yet - open command pallet using :kbd:`Ctrl-Shift-P` and select ``Cmake: Configure``
+- In the kit selection choose ``[Unspecified]`` unless you have more specific needs.
+- If you see error "CMakeList.txt was not found in the root of folder cutter" chose ``Locate`` and specify the path to ``cutter/src/CMakeLists.txt``
+- :kbd:`Ctrl-Shift-P`/``CMake: Edit CMake Cache``, find the line ``CUTTER_USE_BUNDLED_RADARE2:BOOL=OFF`` and change it to ON.
+- Download Qt type visualizer for VS debugger from
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:97]: Debugger visualizer instructions moved `Building, Running, Debugging` section

File: docs/source/developers-docs/ide-setup.rst [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+CLion
+-----
+`CLion <https://www.jetbrains.com/clion/>`_ is a C and C++ IDE from the popular software development tool maker - JetBrains.
+
+
+Pros and Cons
+~~~~~~~~~~~~~
+
+- Medium amount of plugins, many first-party plugins made by JetBrains for their IntelliJ based IDE family
+- There is no free version
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:137]: That's very limited. Not everyone is a student. In that case you might as well call "30 day trial" a free versions. All other IDEs mentioned in the page have free version that can be used by anyone.

File: docs/source/developers-docs/ide-setup.rst [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+
+    sys.path.insert(0, '/path/to/folder/with/pretty_printer_scripts')
+    from qt import register_qt_printers
+    register_qt_printers (None)
+
+    end
+    set print pretty 1
+
+
+CLion
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/developers-docs/ide-setup.rst:128]: Added link to CLion documentation for configuring Qt type printers.

**Pull request merged**